osu.Game.Tests/Visual/SongSelectV2/TestSceneBeatmapCarouselUpdateHandling.cs

```diff
@@ -139,6 +139,65 @@ public void TestSelectionHeldDifficultyOnlineIDChanged()
             AddAssert("visible panel is updateable beatmap", () => GetSelectedPanel()?.Item?.Model, () => Is.EqualTo(baseTestBeatmap.Beatmaps[0]));
         }
 
+        [Test] // Checks that we don't crash if there exists a difficulty with the same name as the selected difficulty.
+        public void TestDifferentDifficultiesWithSameName()
+        {
+            SelectNextGroup();
+
+            WaitForSelection(1, 0);
+            AddAssert("selection is updateable beatmap", () => Carousel.CurrentSelection, () => Is.EqualTo(baseTestBeatmap.Beatmaps[0]));
+            AddAssert("visible panel is updateable beatmap", () => GetSelectedPanel()?.Item?.Model, () => Is.EqualTo(baseTestBeatmap.Beatmaps[0]));
+
+            // This scenario is pretty much silly, but it's possible to do.
```

Comment:
Are you saying that the issue reporter is incorrectly stating what they did? They say that they renamed the difficulty before adding it, but you're saying there needs to be two difficulties with the same name?

Reply:
The reporter's scenario is covered by the second test, which adds another difficulty while keeping the same online ID. The reporter could be asked to confirm that this is exactly the case, but it's the only way to explain this failure so I didn't bother to.